lib/ui/painting.dart

```diff
@@ -1663,6 +1666,20 @@ class Image {
   /// The number of image pixels along the image's vertical axis.
   final int height;
 
+  /// A callback that is invoked to report an object creation.
+  ///
+  /// It's preferred to use [MemoryAllocations] in flutter/foundation.dart 
+  /// than to use [onCreate] directly because [MemoryAllocations]
+  /// allows multiple callbacks.
+  static void Function(Object)? onCreate;
+
+  /// A callback that is invoked to report the object disposal.
+  ///
+  /// It's preferred to use [MemoryAllocations] in flutter/foundation.dart 
+  /// than to use [onDispose] directly because [MemoryAllocations]
+  /// allows multiple callbacks.
+  static void Function(Object)? onDispose;
```

Comment:
Is this intended to be use in production? If not, consider prepending `debug` in the name of the API, e.g. `debugOnDispose`.

(applies to `onCreate` as well)

Reply:
It will be off by default in production, but inableable.